runtime/interpreter/value.go

```diff
@@ -18274,13 +18420,17 @@ func (v *DictionaryValue) GetMember(
 						false,
 						nil,
 						nil,
+						false, // value is an element of parent container because it is returned from iterator.
 					)
 			},
 		)
 
 	case "values":
 
-		iterator, err := v.dictionary.Iterator()
+		valueComparator := newValueComparator(interpreter, locationRange)
+		hashInputProvider := newHashInputProvider(interpreter, locationRange)
+
+		iterator, err := v.dictionary.Iterator(valueComparator, hashInputProvider)
```

Comment:
We have to use a mutable iterator here, because the values that are returned by the iterator might get mutated, so need to have the parent notification callback set up, correct?

Reply:
Correct.  :+1: